STACpopulator/populator_base.py

```diff
@@ -142,23 +142,39 @@ def create_stac_collection(self) -> dict[str, Any]:
     def publish_stac_collection(self, collection_data: dict[str, Any]) -> None:
         post_stac_collection(self.stac_host, collection_data, self.update, session=self._session)
 
+    def open_error_file(self) -> TextIO:
```

Comment:
This is essentially what `logging.FileHandler` does, but it properly handles thread-safe operations as well.